### PR TITLE
Fix/add documentation comments on StoreGateway service implementations

### PR DESCRIPTION
**Pull request by aknuds1**
#### What this PR does
Fix and add documentation comments on `storepb.StoreGatewayServer` implementation `StoreGateway` and `storepb.StoreServer` implementation `BucketStores`.

#### Checklist

- [na] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/storegateway/bucket_stores.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func (u *BucketStores) syncUsersBlocks(ctx context.Context, f func(context.Conte
 	return errs.Err()
 }
 
-// Series makes a series request to the underlying user bucket store.
+// Series implements the StoreGateway proto service, making a series request to the underlying user bucket store.
```
Comment (pstibrany) [inline pkg/storegateway/bucket_stores.go:296]: It implements StoreServer interface. We can be explicit, and let type linking in comments do the job:

```suggestion
// Series implements storepb.StoreServer interface, making a series request to the underlying user bucket store.
```

**Review by pstibrany — APPROVED** [review]

File: pkg/storegateway/bucket_stores.go [hunk 2]
```diff
@@ -314,7 +314,7 @@ func (u *BucketStores) Series(req *storepb.SeriesRequest, srv storepb.Store_Seri
 	})
 }
 
-// LabelNames implements the Storegateway proto service.
+// LabelNames implements the StoreGateway proto service.
```
Comment (pstibrany) [inline pkg/storegateway/bucket_stores.go:317]: Same comment.

Comment (aknuds1) [reply pkg/storegateway/bucket_stores.go]: Fixed.

File: pkg/storegateway/bucket_stores.go [hunk 3]
```diff
@@ -332,7 +332,7 @@ func (u *BucketStores) LabelNames(ctx context.Context, req *storepb.LabelNamesRe
 	return store.LabelNames(ctx, req)
 }
 
-// LabelValues implements the Storegateway proto service.
+// LabelValues implements the StoreGateway proto service.
```
[the remark this answers is not in the document]
Comment (aknuds1) [reply pkg/storegateway/bucket_stores.go]: Fixed.

**Review by pstibrany — APPROVED** [review]
Thank you.

**Pull request merged**